### PR TITLE
feat(sdk): support sql parameters in interactive questions

### LINKED ISSUE #50493: Support providing parameter values in InteractiveQuestion in the sdk
We had a [Slack discussion thread](https://metaboat.slack.com/archives/C063Q3F1HPF/p1721234527748689) about implementing `parameterValues` in interactive questions back in July. Back then, we decided to close https://github.com/metabase/metabase/pull/45664 because of [a lack of drill-down support](https://metaboat.slack.com/archives/C063Q3F1HPF/p1721247361723199?thread_ts=1721234527.748689&cid=C063Q3F1HPF) in SQL questions.

Now that the team has implemented SQL drill-downs into the product, we can implement parameter values in InteractiveQuestion. We should be able to repurpose https://github.com/metabase/metabase/pull/45664 with some small fixes to make it work with the current codebase.

### PR DESCRIPTION
**Pull request by heypoom**
Closes https://github.com/metabase/metabase/issues/50493

### Description

Adds the `initialSqlParameters` to interactive questions for feature parity with static questions.

### How to verify

- Render an interactive question with `initialSqlParameters`, using a SQL question for the question id
- The SQL parameters should be passed to the SQL question

### Demos

CleanShot 2567-12-04 at 15 10 01@2x

### Checklist

- [x] Tests have been added/updated to cover changes in this PR - E2E

### CODE REVIEW

File: docs/embedding/sdk/questions.md
Comment (heypoom) [inline docs/embedding/sdk/questions.md:85]: Note that `initialSqlParameters` is also exposed for static questions in https://github.com/metabase/metabase/pull/50656 -- I'm merging that PR later today.

File: e2e/test-component/scenarios/embedding-sdk/interactive-question-native.cy.spec.tsx
Comment (heypoom) [inline e2e/test-component/scenarios/embedding-sdk/interactive-question-native.cy.spec.tsx:51]: As `ID` is a primary key in the sample database, we can safely assume that `{ID: ORDERS_ID}` should give us one row with the specified ID.

File: enterprise/frontend/src/embedding-sdk/hooks/private/use-load-question.ts
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/hooks/private/use-load-question.ts:79]: When passing `initialSqlParameters={{ID: "12"}}`, this may cause an infinite re-render because the object reference is created every render. This makes it so that we only update the `loadQuestion` callback's reference if the SQL parameter actually changed.

File: enterprise/frontend/src/embedding-sdk/lib/interactive-question/run-question-on-load.ts
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/lib/interactive-question/run-question-on-load.ts:45]: This gets the template tag parameters from the card, then apply it with `setParameters` internally, so the question object is aware of what parameters are there. This allows us to use `setParameterValues` to assign the SQL parameter values later.

File: enterprise/frontend/src/embedding-sdk/lib/interactive-question/run-question-on-load.ts
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/lib/interactive-question/run-question-on-load.ts:48]: The SQL parameter values are passed as strings, as we already denoted the type in the native query editor.

File: frontend/src/metabase/query_builder/actions/core/updateQuestion.ts
Comment (heypoom) [inline frontend/src/metabase/query_builder/actions/core/updateQuestion.ts:131]: This is the exact same logic, just extracted into `syncCardParametersWithTemplateTags` so I can reuse it everywhere else including in the SDK.

File: frontend/src/metabase/query_builder/actions/core/utils.ts
Comment (heypoom) [inline frontend/src/metabase/query_builder/actions/core/utils.ts:84]: This is from the existing diff, just extracted into a new function.

https://github.com/metabase/metabase/pull/50728/files#diff-feedbe46976731ce62d4f714452c04eff7e9d8f6a76ebca1ea2fd1f8864ab780L129-L133

File: docs/embedding/sdk/questions.md
Comment (oisincoveney) [inline docs/embedding/sdk/questions.md:40]: If we have a combined section for static and interactive questions, I don't think we need to specify this. I think it's already implied

**Review by oisincoveney — APPROVED** [review]
one lint error, but I don't have any notes. looks good. just make sure to get a review from team-querying as well

File: docs/embedding/sdk/questions.md
Comment (heypoom) [reply docs/embedding/sdk/questions.md:40]: @oisincoveney This is in the "Embedding a static question" so it's a bit weird. Let me actually separate this into a separate section, cc @jeff-bruemmer

Comment (heypoom) [reply docs/embedding/sdk/questions.md]: updated in https://github.com/metabase/metabase/pull/50728/commits/ee5aa7953a163849e0812fd5b512627caa9c4dbd

File: e2e/test-component/scenarios/embedding-sdk/interactive-question-native.cy.spec.tsx
Comment (WiNloSt) [reply e2e/test-component/scenarios/embedding-sdk/interactive-question-native.cy.spec.tsx:51]: I wonder if asserting the rendered UI is correct is better. To be fair, I'd feel more confident if we assert the UI, not the API response.

But I see the problem with going with the UI route is that I don't think we can see the amount of rows the API is returning.

**Review by WiNloSt — APPROVED** [review]

File: e2e/test-component/scenarios/embedding-sdk/interactive-question-native.cy.spec.tsx
Comment (kamilmielnik) [inline e2e/test-component/scenarios/embedding-sdk/interactive-question-native.cy.spec.tsx:39]: nit: we usually keep the setup in `beforeEach`

File: frontend/src/metabase/query_builder/actions/core/utils.ts
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/actions/core/utils.ts:74]: nit: I wonder whether this function should be part of `Question` class (since it receives a `Question` and returns a `Question`)

File: enterprise/frontend/src/embedding-sdk/lib/interactive-question/run-question-on-load.ts
Comment (kamilmielnik) [inline enterprise/frontend/src/embedding-sdk/lib/interactive-question/run-question-on-load.ts:48]: nit: we usually import the entire underscore
```suggestion
      ? _.mapObject(initialSqlParameters, String)
```

**Review by kamilmielnik — APPROVED** [review]
Looks good!

Please address [this comment](https://github.com/metabase/metabase/pull/50728#discussion_r1875513655) from Kelvin before merging :pray:

File: docs/embedding/sdk/questions.md
Comment (heypoom) [reply docs/embedding/sdk/questions.md:84]: @WiNloSt It looks like this is for the Metabase docs site; I see @jeff-bruemmer adding this to the code snippets originally. I also see this in other markdown files as well, but I'm not exactly sure where we would need the `{% raw %}` tag.

File: enterprise/frontend/src/embedding-sdk/hooks/private/use-load-question.ts
Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/hooks/private/use-load-question.ts]: @WiNloSt Good call, made it key-order independent in https://github.com/metabase/metabase/pull/50728/commits/171d2a525825e5c310d7eb3de3be3806de4b313d.

PS. I went with creating a dependency key that is key-order independent instead of using isEqual, as that would require using `usePrevious` and `isEqual` to compare other dependencies as well.

File: e2e/test-component/scenarios/embedding-sdk/interactive-question-native.cy.spec.tsx
Comment (heypoom) [reply e2e/test-component/scenarios/embedding-sdk/interactive-question-native.cy.spec.tsx:51]: We should be able count the number of divs from the table root to get the row count too, but yes it's a little trickier to assert the UI. I just noticed we have some helpers in `assertTestData` though, so I can try to use that for assertions.

https://github.com/metabase/metabase/blob/6694870932830ff36964edd8b2c4555b81ddeda8/e2e/support/helpers/e2e-ui-elements-helpers.js#L310-L317

Comment (WiNloSt) [reply e2e/test-component/scenarios/embedding-sdk/interactive-question-native.cy.spec.tsx:51]: well yeah, let's see how it goes, if we have a good argument to keep asserting the API then I'm also cool with that.

File: enterprise/frontend/src/embedding-sdk/lib/interactive-question/run-question-on-load.ts
Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/lib/interactive-question/run-question-on-load.ts]: Updated in https://github.com/metabase/metabase/pull/50728/commits/d291c7fafbbdbb82b7cedc5291d7f4cff2dc4215

PS. I just noticed we don't have plugins that tree-shakes underscore usage from the bundle, but I suppose that's the least of our worries considering there's so much in the bundle

File: e2e/test-component/scenarios/embedding-sdk/interactive-question-native.cy.spec.tsx
Comment (heypoom) [reply e2e/test-component/scenarios/embedding-sdk/interactive-question-native.cy.spec.tsx:37]: Updated in https://github.com/metabase/metabase/pull/50728/commits/d291c7fafbbdbb82b7cedc5291d7f4cff2dc4215, thanks!

File: frontend/src/metabase/query_builder/actions/core/utils.ts
Comment (heypoom) [reply frontend/src/metabase/query_builder/actions/core/utils.ts]: Good idea, fixed in https://github.com/metabase/metabase/pull/50728/commits/a8e3ee751278a50a7d26cd0e5b275060f3a0e94e

File: e2e/test-component/scenarios/embedding-sdk/interactive-question-native.cy.spec.tsx
Comment (heypoom) [reply e2e/test-component/scenarios/embedding-sdk/interactive-question-native.cy.spec.tsx]: @WiNloSt fixed in https://github.com/metabase/metabase/pull/50728/commits/c6ac1e7d96a54e08d8805e7a8a06d8373d828d3c, now it's asserting the table's UI instead.

File: docs/embedding/sdk/questions.md
Comment (jeff-bruemmer) [reply docs/embedding/sdk/questions.md:84]: Correct. It's for code blocks that contain curly braces, e.g., `{{ var_name }}`. Jekyll interprets mustached text as variables, so the raw tags tell Jekyll to preserve the code as is.

https://shopify.dev/docs/api/liquid/tags#raw

**Comment by heypoom** [comment]
@metabase-bot backport release-x.51.x

**Pull request merged**
